src/main/java/org/opensearch/security/auth/limiting/AbstractRateLimiter.java

```diff
@@ -47,6 +53,19 @@ public AbstractRateLimiter(Settings settings, Path configPath, Class<ClientIdTyp
     @Override
     public abstract void onAuthFailure(InetAddress remoteAddress, AuthCredentials authCredentials, Object request);
 
+    @Override
+    public WildcardMatcher getIgnoreHostsMatcher() {
+        if (this.ignoreHostMatcher != null) {
+            return this.ignoreHostMatcher;
+        }
```

Comment:
Would ignoreHostMatcher get re-initialized on updates/changes to rate limiting config?

Reply:
Yes it will and I added a test that verifies that case. The test uses the `PATCH /_plugins/security/api/securityconfig` API to replace ignore_hosts to add an additional value.

1. Instances of an AuthFailureListener are created in [DynamicConfigModelV7.createAuthFailureListeners](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/securityconf/DynamicConfigModelV7.java#L403-L409)
2. `DynamicConfigModelV7.createAuthFailureListeners` is called from within `DynamicConfigModelV7.buildAAA`
3. `buildAAA` is called within the [constructor](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/securityconf/DynamicConfigModelV7.java#L102) of DynamicConfigModelV7
4. The constructor is called during a dynamic config update [here](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/securityconf/DynamicConfigFactory.java#L279)